olive/systems/azureml/aml_system.py

```diff
@@ -130,79 +146,79 @@ def run_pass(
-        if model_json["config"].get("model_path"):
-            model_resource_path = create_resource_path(model_json["config"]["model_path"])
-            asset_type = RESOURCE_TYPE_TO_ASSET_TYPE[model_resource_path.type]
-
-            if model_resource_path.type in AZUREML_RESOURCE_TYPES:
-                # ensure that the model is in the same workspace as the system
-                model_workspace_config = model_resource_path.config.azureml_client.get_workspace_config()
-                system_workspace_config = self.azureml_client_config.get_workspace_config()
-                for key in model_workspace_config:
-                    if model_workspace_config[key] != system_workspace_config[key]:
-                        raise ValueError(
-                            f"Model workspace {model_workspace_config} is different from system workspace"
-                            f" {system_workspace_config}."
-                        )
-
-            if model_resource_path.type == ResourceType.AzureMLJobOutput:
-                # there is no direct way to use the output of a job as input to another job
-                # so we create a dummy aml model and use it as input
-                ml_client = self.azureml_client_config.create_client()
-
-                # create aml model
-                logger.debug(f"Creating aml model for job output {model_resource_path}")
-                aml_model = retry_func(
-                    ml_client.models.create_or_update,
-                    [
-                        Model(
-                            path=model_resource_path.get_path(),
-                            name="olive-backend-model",
-                            description="Model created by Olive backend. Ignore this model.",
-                            type=AssetTypes.CUSTOM_MODEL,
-                        )
-                    ],
-                    max_tries=self.azureml_client_config.max_operation_retries,
-                    delay=self.azureml_client_config.operation_retry_interval,
-                    exceptions=ServiceResponseError,
-                )
-                model_resource_path = create_resource_path(
-                    AzureMLModel(
-                        {
-                            "azureml_client": self.azureml_client_config,
-                            "name": aml_model.name,
-                            "version": aml_model.version,
-                        }
+            # create aml model
+            logger.debug(f"Creating aml model for job output {model_resource_path}")
+            aml_model = retry_func(
+                ml_client.models.create_or_update,
+                [
+                    Model(
+                        path=model_resource_path.get_path(),
+                        name="olive-backend-model",
+                        description="Model created by Olive backend. Ignore this model.",
+                        type=AssetTypes.CUSTOM_MODEL,
                     )
```

Comment:
why do we need create the model?

Reply:
This is a workaround to use job outputs as aml inputs. 
![image](https://github.com/microsoft/Olive/assets/94929125/83b01e3e-9a6f-43a9-a2cd-8873c863c41f)

This comment describes why we do it. I have asked the aml sdk team for help since aml jobs are supposed to be supported paths https://learn.microsoft.com/en-us/azure/machine-learning/how-to-manage-models?view=azureml-api-2&tabs=cli%2Cuse-local#supported-paths but it doesn't work as expected. They haven't replied yet so we are using this as a workaround for now.

Reply:
@jambayk , need we add TODO here?